File: greedy/cpp/b1052.cpp
#include <iostream>

int N, K;

int check_bottles(int nums) {
	int cnt = 0;
	while (nums) {
		if (nums % 2) {
			cnt++;
		}
		nums = nums / 2;
	}
	return cnt;
}

int main() {
	std::cin >> N >> K;
	int result = 0;
	int cnt = 1;
	while (check_bottles(N) > K) {
		if (N & 1) {
			N += 1;
			result += cnt;
		}
		N = N >> 1;
		cnt *= 2;
	}
	std::cout << result << std::endl;
}

File: greedy/python/b1052.py
def check(nums):
    cnt = 0
    while nums:
        if nums % 2:
            cnt += 1
        nums = nums >> 1
    return cnt

N, K = map(int, input().split())

buy = 0
times = 1
while check(N) > K:
    if N & 1:
        buy += times
        N += 1
    N = N >> 1
    times *= 2

print(buy)